store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -9555,6 +9555,66 @@ public Document createDocument(OperationContext octxt, int folderId, ParsedDocum
         }
     }
 
+    public Document createDocument(OperationContext octxt, int folderId, ParsedDocument pd, MailItem.Type type,
```

Comment:
Use method overloading instead of creating the same duplicate method.

Reply:
This is method overloading only, introduced one new parameter at end as nodeId.
>public Document createDocument(OperationContext octxt, int folderId, String filename, String mimeType,
            String author, String description, InputStream data)

>public Document createDocument(OperationContext octxt, int folderId, ParsedDocument pd, MailItem.Type type,
            int flags, MailItem parent, CustomMetadata custom, boolean indexing, String nodeId)

Reply:
Yes, you can achieve the same thing without having duplicate methods. like 

```
public Document createDocument(OperationContext octxt, int folderId, ParsedDocument pd, MailItem.Type type,
             int flags, MailItem parent, CustomMetadata custom, boolean indexing) {
             createDocument(OperationContext octxt, int folderId, ParsedDocument pd, MailItem.Type type,
             int flags, MailItem parent, CustomMetadata custom, boolean indexing, null);
             }`
```

and method signature will be for the existing method with the addition of a new parameter.
```
public Document createDocument(OperationContext octxt, int folderId, ParsedDocument pd, MailItem.Type type,
            int flags, MailItem parent, CustomMetadata custom, boolean indexing, String nodeId)
```

Reply:
Done